code/API_definitions/device-data-volume-subscriptions.yaml

```diff
@@ -0,0 +1,1259 @@
+openapi: 3.0.3
+info:
+  title: Device Data Volume Subscriptions
+  description: |
+    This API enables customers to subscribe to events related to data volume usage thresholds on a device. It supports both request and subscription modes to track remaining data volume.
+
+    # Introduction
+
+    ## Device Data Volume Subscriptions
+
+    API consumers can receive notifications regarding data volume usage thresholds, such as when the remaining data volume reaches 50%, 75%, or 90% of the user's allocated data plan. This is useful for managing data usage and avoiding overage fees.
+
+    # Relevant terms and definitions
+
+    * **Device**: A device refers to any physical entity capable of network communication, identified via an IP address, phone number, or Network Access Identifier.
+
+    # API Functionality
+
+    The API exposes the following capabilities:
+
+    ## Device Data Volume Subscriptions
+
+    These endpoints allow management of subscriptions to data volume usage events. The CAMARA subscription model follows the CloudEvents specification.
+
+    When subscribing, it is mandatory to provide the event `types` related to data volume thresholds you wish to track.
+
+    The following event `types` are managed for this API:
+      - `org.camaraproject.device-data-volume-subscriptions.v0.data-50-percent`: Event triggered when 50% of the data plan is consumed.
+      - `org.camaraproject.device-data-volume-subscriptions.v0.data-75-percent`: Event triggered when 75% of the data plan is consumed.
+      - `org.camaraproject.device-data-volume-subscriptions.v0.data-90-percent`: Event triggered when 90% of the data plan is consumed.
+      - `org.camaraproject.device-data-volume-subscriptions.v0.data-exceeded`: Event triggered when the data plan is exceeded.
+
+    Note: Additionally, the event `org.camaraproject.device-data-volume-subscriptions.v0.subscription-ends` is sent when the subscription ends. This notification does not require a dedicated subscription.
+
+    ### Notifications callback
+
+    This endpoint describes the event notification received on the subscriber's callback URL when the event occurs. Developers should provide a callback URL to receive notifications regarding data volume usage events from the service provider.
+
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: wip
+  x-camara-commonalities: 0.4.0
+
+servers:
+  - url: "{apiRoot}/device-data-volume-subscriptions/v0"
```

Comment:
should be {apiRoot}/device-data-volume-subscriptions/wip no?

Reply:
{apiRoot}/device-data-volume-subscriptions/vwip

Reply:
Yes :) `vwip ` in the url

Reply:
Done.